src/sqlancer/influxdb/InfluxDBProvider.java

```diff
@@ -0,0 +1,112 @@
+package sqlancer.influxdb;
+
+import com.google.auto.service.AutoService;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import sqlancer.*;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLQueryProvider;
+import sqlancer.influxdb.gen.InfluxDBCreateDatabaseGenerator;
+import sqlancer.influxdb.gen.InfluxDBWritePointGenerator;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Properties;
+
+@AutoService(DatabaseProvider.class)
+public class InfluxDBProvider extends SQLProviderAdapter<InfluxDBProvider.InfluxDBGlobalState, InfluxDBOptions> {
+
+    public InfluxDBProvider() {
+        super(InfluxDBGlobalState.class, InfluxDBOptions.class);
+    }
+
+    public enum Action implements AbstractAction<InfluxDBGlobalState> {
+        WRITE_POINT(sqlancer.influxdb.gen.InfluxDBInsertGenerator::getQuery),
+        CREATE_DATABASE(sqlancer.influxdb.gen.InfluxDBInsertGenerator::getQuery),
+        InfluxDBWritePointGenerator();
+
+        private final SQLQueryProvider<InfluxDBGlobalState> sqlQueryProvider;
+
+        Action(SQLQueryProvider<InfluxDBGlobalState> sqlQueryProvider) {
+            this.sqlQueryProvider = sqlQueryProvider;
+        }
+
+        @Override
+        public SQLQueryAdapter getQuery(InfluxDBGlobalState state) throws Exception {
+            return sqlQueryProvider.getQuery(state);
+        }
+    }
+
+    private static int mapActions(InfluxDBGlobalState globalState, Action a) {
+        Randomly r = globalState.getRandomly();
+        switch (a) {
+            case WRITE_POINT:
+                return r.getInteger(0, globalState.getOptions().getMaxNumberWrites());
```

Comment:
We have an option related to `INSERT`s. Is there some overlap with this method? If not, do we use it for `INSERT`s?

https://github.com/sqlancer/sqlancer/blob/04f0b49ccbcee1054ef93f27b0055f2aa976ca05/src/sqlancer/MainOptions.java#L30